wire.go

```diff
@@ -132,3 +133,22 @@ func Value(interface{}) ProvidedValue {
 func InterfaceValue(typ interface{}, x interface{}) ProvidedValue {
 	return ProvidedValue{}
 }
+
+// StructFields is a collection of the fields from a struct.
+type StructFields struct{}
+
+// FieldsOf extracts the structType into a list of types specified by fieldNames.
+// All types should be provided in the same provider set. The structType argument
+// must be a pointer to the struct it wishes to reference.
+//
+// Example:
+//
+//  type S struct {
+//  	MyFoo *Foo
+//  	MyBar *Bar
+//  }
+//  func NewStruct() *S { /* ... */ }
+//  var Set = wire.NewSet(wire.FieldsOf(new(S), "Foo", "Bar"))
```

Comment:
IIUC this should be:

```suggestion
//  var Set = wire.NewSet(wire.FieldsOf(new(S), "MyFoo", "MyBar"))
```